redbot/core/core_commands.py

```diff
@@ -2060,43 +2060,59 @@ async def slash_disable(
             )
         )
 
-    @slash.command(name="enablecog")
+    @slash.command(name="enablecog", require_var_positional=True)
     @commands.max_concurrency(1, wait=True)
-    async def slash_enablecog(self, ctx: commands.Context, cog_name: str):
+    async def slash_enablecog(self, ctx: commands.Context, *cog_names: str):
         """Marks all application commands in a cog as being enabled, allowing them to be added to the bot.
 
         See a list of cogs with application commands with `[p]slash list`.
 
         This command does NOT sync the enabled commands with Discord, that must be done manually with `[p]slash sync` for commands to appear in users' clients.
 
         **Arguments:**
-            - `<cog_name>` - The cog to enable commands from. This argument is case sensitive.
+            - `<cog_names>` - The cogs to enable commands from. This argument is case sensitive.
         """
         enabled_commands = await self.bot.list_enabled_app_commands()
         to_add_slash = []
         to_add_message = []
         to_add_user = []
 
+        successful_cogs = set()
```

Comment:
This variable doesn't actually do anything because there is no way for an individual cog to be unsuccessful how you have it here. The command currently enables all cogs requested or none of the cogs requested, if any would cause it to fail.

It makes no special casing for if an *individual* cog has no commands or puts the tree over the command limit, only if *all* cogs requested have no commands or put the tree over the limit.

In order to allow the command to enable only a subset of cogs, while still either enabling or not enabling each individual cog, you would probably want to loop over a helper method of the current code for this method to do that, rather than trying to integrate logic for multiple cogs into each step at the same time. The existing code is supposed to be a single atomic step, which limits what can be done when integrating support for multiple cogs.